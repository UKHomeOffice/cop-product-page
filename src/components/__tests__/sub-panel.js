import React from "react";
import SubPanel from "../sub-panel";
import { create } from "react-test-renderer";
import renderer from "react-test-renderer";

describe("Header", () => {
  it("Render correctly", async () => {
    const tree = renderer
      .create(<SubPanel/>);
    expect(tree).toMatchSnapshot();
  });

  it("button onclick render correctly", async () => {
    const tree = create(<SubPanel/>);
    const instance = tree.root;

    const button = instance.findByProps({id: 'boxclose'});
    button.props.onClick();
    expect(tree).toMatchSnapshot();
  });


});

