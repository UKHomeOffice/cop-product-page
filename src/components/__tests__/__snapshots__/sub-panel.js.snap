// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="govuk-panel"
  id="sub-panel"
  style={
    Object {
      "background": "#bbd4fe",
      "margin": "0 auto",
      "paddingTop": "10px",
    }
  }
>
  <div
    className="govuk-width-container"
  >
    <a
      className="boxclose govuk-link"
      id="boxclose"
      onClick={[Function]}
      onKeyPress={[Function]}
    />
    <div
      className="govuk-grid-row"
    >
      <div
        className="govuk-grid-column-two-thirds"
      >
        <p
          className="govuk-body-m"
          style={
            Object {
              "marginBottom": "0px",
            }
          }
        >
          <span
            className="govuk-!-font-weight-bold"
          >
            Mandatory Declarations (Man Dec)
          </span>
           submission deadline is 
          <span
            className="govuk-!-font-weight-bold"
          >
            31 April 2020
          </span>
          <br />
          <a
            className="govuk-link govuk-!-font-weight-bold"
            href="https://www.cop.homeoffice.gov.uk/"
          >
            Sign-in
          </a>
           to COP to fill the form, or 
          <a
            className="govuk-link govuk-!-font-weight-bold"
            href="https://training.cop.homeoffice.gov.uk/mandec.html#mandatory-declarations-man-dec"
          >
            find out more about the Man Dec.
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
